doc/src/content/xdocs/spec.xml

```diff
@@ -273,60 +273,125 @@
+        "symbols": ["a", "b"]
+      }
+    }, {
+      "name": "explicitNamespace",
+      "type": {
+        "type": "fixed",
+        "name": "simple",
+        "namespace": "explicit",
+        "doc": "A simple name (attribute) and a namespace (attribute); the fullname is 'explicit.Simple' (this is a different type than of the 'inheritNull' field)",
+        "size": 12
+      }
+    }, {
+      "name": "fullName",
+      "type": {
+        "type": "record",
+        "name": "a.full.Name",
+        "namespace": "ignored",
+        "doc": "A name attribute with a fullname, so the namespace attribute is ignored. The fullname is 'a.full.Name', and the namespace is 'a.full'.",
+        "fields": [
+          {
+            "name": "inheritNamespace",
+            "type": {
+              "type": "enum",
+              "name": "Understanding",
+              "doc": "A simple name (attribute) and no namespace attribute: inherit the namespace of the enclosing type 'a.full.Name'. The fullname is 'a.full.Understanding'.",
+              "symbols": ["d", "e"]
+            }
+          }
+        ]
+      }
+    }
+  ]
+}
+        </source>
+        <p>The fullname of a record, enum or fixed definition is determined by
+          the required `name` and optional `namespace` attributes like this:</p>
+        <ul>
+          <li>A fullname is specified. If the name specified contains
+            a dot, then it is assumed to be a fullname, and <em>any
+              namespace also specified is ignored</em>. For example,
+            use <code>"name": "org.foo.X"</code> to indicate the
+            fullname <code>org.foo.X</code>.
+          </li>
+          <li>A simple name (a name that contains no dots) and namespace are
+            both specified. For example, one might use
+            <code>"name": "X", "namespace": "org.foo"</code> to indicate the
+            fullname <code>org.foo.X</code>.
+          </li>
+          <li>A simple name only is specified (a name that contains no dots).
+            In this case the namespace is taken from the most tightly enclosing
+            named schema or protocol, and the fullname is constructed from that
+            namespace and the name. For example, if <code>"name": "X"</code> is
+            specified, and this occurs within a field of the record definition
+            of <code>org.foo.Y</code>, then the fullname is <code>org.foo.X</code>.
+            This also happens if there is no enclosing namespace (i.e., the
+            enclosing schema definition has the null namespace).
+          </li>
+        </ul>
+        <p>References to previously defined names are as in the latter
+          two cases above: if they contain a dot they are a fullname, if
+          they do not contain a dot, the namespace is the namespace of
+          the enclosing definition.
+        </p>
+        <p>Primitive type names have no namespace and their names may
```

Comment:
Ugh -- this is another problem, isn't it?  At our company we just ran into the inconsistent treatment of a record named `"record"` (a non-primitive type, OK in Java, forbidden in Python). I'll bring this up on the mailing list!

Reply:
https://lists.apache.org/thread/vmx2ocyvt6nbrff9lsdt4plyx5foyzcb